packages/amplify-e2e-tests/src/__tests__/auth/admin-api.test.ts

```diff
@@ -0,0 +1,177 @@
+import {
+  addAuthWithGroupsAndAdminAPI,
+  addUserToGroup,
+  amplifyPushAuth,
+  configureAmplify,
+  createNewProjectDir,
+  deleteProject,
+  deleteProjectDir,
+  getProjectMeta,
+  getUserPoolId,
+  initJSProjectWithProfile,
+  invokeFunction,
+  setupUser,
+} from '@aws-amplify/amplify-e2e-core';
+import { $TSAny } from '@aws-amplify/amplify-cli-core';
+
+describe('auth admin api tests', () => {
+  let projRoot: string;
+  let adminLambdaName: string;
+  let adminLambdaRegion: string;
+
+  const callAdminLambda = async (request: $TSAny): Promise<$TSAny> => {
+    if (!request.requestContext) {
+      // Mock authorization necessary for admin API
+      request.requestContext = {
+        authorizer: {
+          claims: {
+            'cognito:groups': 'Admins',
+          },
+        },
+      };
+    }
+    const lambdaResponse = await invokeFunction(adminLambdaName, JSON.stringify(request), adminLambdaRegion);
+    return JSON.parse(lambdaResponse.Payload.toString());
+  };
+
+  beforeAll(async () => {
+    projRoot = await createNewProjectDir('auth');
+    await initJSProjectWithProfile(projRoot);
+    await addAuthWithGroupsAndAdminAPI(projRoot);
+    await amplifyPushAuth(projRoot);
+
+    const meta = getProjectMeta(projRoot);
+    const adminLambdaOutputs = Object.keys(meta.function).map((key) => meta.function[key])[0].output;
+    adminLambdaName = adminLambdaOutputs.Name;
+    adminLambdaRegion = adminLambdaOutputs.Region;
+
+    const userPoolId = getUserPoolId(projRoot);
+    await configureAmplify(projRoot);
+
+    for (let i = 0; i < 5; i++) {
+      const username = `testUser${i}`;
+      const password = `Password12#${i}`;
+      await setupUser(userPoolId, username, password, 'Users', adminLambdaRegion);
+    }
+
+    for (let i = 0; i < 5; i++) {
```

Comment:
Why don't we only use one loop to set both user and admin?

Reply:
it might have spared a line or two but it would obscure the intent a bit, i.e. that we want to create two (rather independent) groups of users. so I'd rather keep this as is.

Reply:
What if we create a function `setUpUserGroup`? 

Then,

```ts
await setupUserGroup({groupName: "Users", toAddUserToGroup: false});
await setupUserGroup({groupName: "Admins", toAddUserToGroup: true})
```

Would it be easier to read?

Reply:
Rather 

```
await setupUsers({ groups: ['Admins', 'Users'], numberOfUsers: 5, namePrefix: 'testAdminUser'} );
await setupUsers({ groups: ['Users'], numberOfUsers: 5, namePrefix: 'testUser'} );
```

But it'd do that when we get second test file that needs this kind of logic, i.e. when we'd need to repeat this pattern. We have couple of places